ddtrace/contrib/internal/dd_trace_api/patch.py

```diff
@@ -0,0 +1,79 @@
+from sys import addaudithook
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+import weakref
+
+import dd_trace_api
+
+import ddtrace
+
+
+_DD_HOOK_NAME = "dd.hook"
+_TRACER_KEY = "Tracer"
+_STUB_TO_REAL = weakref.WeakKeyDictionary()
+_STUB_TO_REAL[dd_trace_api.tracer] = ddtrace.tracer
+
+
+def _proxy_span_arguments(args: List, kwargs: Dict) -> Tuple[List, Dict]:
+    """Convert all dd_trace_api.Span objects in the args/kwargs collections to their held ddtrace.Span objects"""
```

Comment:
I would do something like that. Comprehensions have also the benefit to be faster.
```suggestion
    def _proxy_span_arguments(args: List, kwargs: Dict) -> Tuple[List, Dict]:
        """Convert all dd_trace_api.Span objects in the args/kwargs collections to their held ddtrace.Span objects"""
        convert = lambda arg: _STUB_TO_REAL[arg] if isinstance(arg, dd_trace_api.Span) else arg
        return [convert(arg) for arg in args], {name: convert(kwarg) for name, kwarg in kwargs.items()}
```